state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1269,87 +1359,108 @@ pub mod pallet {
-				log_or_panic!("CCM egress scheduling should never fail.");
-			}
+			let execute_at = frame_system::Pallet::<T>::block_number() + SWAP_DELAY_BLOCKS.into();
+
+			SwapQueue::<T>::append(
+				execute_at,
+				Swap::new(
+					swap_id,
+					swap_request_id,
+					input_asset,
+					output_asset,
+					input_amount,
+					refund_params,
+					swap_type,
+				),
+			);
+
+			Self::deposit_event(Event::<T>::SwapScheduled {
+				swap_request_id,
+				swap_id,
+				input_amount,
+				swap_type,
+				execute_at,
+			});
+
+			swap_id
 		}
 	}
 
-	impl<T: Config> SwapDepositHandler for Pallet<T> {
+	impl<T: Config> SwapRequestHandler for Pallet<T> {
 		type AccountId = T::AccountId;
 
-		/// Callback function to kick off the swapping process after a successful deposit.
-		fn schedule_swap_from_channel(
-			deposit_address: ForeignChainAddress,
-			deposit_block_height: u64,
-			from: Asset,
-			to: Asset,
-			amount: AssetAmount,
-			destination_address: ForeignChainAddress,
-			broker_commission: Beneficiaries<Self::AccountId>,
+		fn init_swap_request(
+			input_asset: Asset,
+			input_amount: AssetAmount,
+			output_asset: Asset,
+			request_type: SwapRequestType,
+			broker_fees: Beneficiaries<Self::AccountId>,
 			refund_params: Option<ChannelRefundParameters>,
-			channel_id: ChannelId,
-		) -> SwapId {
-			// Permill maxes out at 100% so this is safe.
-			let fee: u128 = Permill::from_parts(
-				broker_commission.iter().fold(0, |acc, entry| acc + entry.bps) as u32 *
-					BASIS_POINTS_PER_MILLION,
-			) * amount;
+			origin: SwapOrigin,
+		) -> Result<SwapRequestId, DispatchError> {
+			let (net_amount, broker_fee) = {
+				// Subtract broker fee:
+
+				// Permill maxes out at 100% so this is safe.
+				let fee: u128 = Permill::from_parts(
+					broker_fees.iter().fold(0, |acc, entry| acc + entry.bps) as u32 *
+						BASIS_POINTS_PER_MILLION,
+				) * input_amount;
+
+				assert!(fee <= input_amount, "Broker fee cannot be more than the amount");
+
+				for Beneficiary { account, bps } in broker_fees {
+					EarnedBrokerFees::<T>::mutate(&account, input_asset, |earned_fees| {
+						earned_fees.saturating_accrue(
+							Permill::from_parts(bps as u32 * BASIS_POINTS_PER_MILLION) *
+								input_amount,
+						)
+					});
+				}
 
-			assert!(fee <= amount, "Broker fee cannot be more than the amount");
+				(input_amount.saturating_sub(fee), fee)
+			};
 
-			let net_amount = amount.saturating_sub(fee);
+			let request_id = SwapRequestIdCounter::<T>::mutate(|id| {
+				id.saturating_accrue(1);
+				*id
+			});
 
-			let encoded_destination_address =
-				T::AddressConverter::to_encoded_address(destination_address.clone());
-			let swap_origin = SwapOrigin::DepositChannel {
-				deposit_address: T::AddressConverter::to_encoded_address(deposit_address),
-				channel_id,
-				deposit_block_height,
+			// Do not limit the maximum swap amount for network fee swaps.
+			let net_amount = if matches!(request_type, SwapRequestType::NetworkFee) {
```

Comment:
Not just network swaps? What about gas fee swaps for example?

Reply:
Added exception for `IngressEgressFee` too.